chrome/content/zotero/itemTree.jsx

```diff
@@ -1740,8 +1741,21 @@ var ItemTree = class ItemTree extends LibraryTree {
 				await Zotero.Items.trashTx(ids);
 			}
 			else if (collectionTreeRow.isCollection()) {
+				let collectionIDs = [collectionTreeRow.ref.id];
+				if (Zotero.Prefs.get('recursiveCollections')) {
+					collectionIDs.push(...collectionTreeRow.ref.getDescendents(false, 'collection').map(c => c.id));
+				}
+
 				await Zotero.DB.executeTransaction(async () => {
-					await collectionTreeRow.ref.removeItems(ids);
+					for (let itemID of ids) {
+						for (let collectionID of collectionIDs) {
+							let item = Zotero.Items.get(itemID);
+							item.removeFromCollection(collectionID);
+							await item.save({
+								skipDateModifiedUpdate: true
+							});
```

Comment:
Move `Zotero.Items.get()` and `item.save()` to the outer loop — only need to save once per item

Reply:
Right, dumb of me. Thanks.